src/adapters/sumer/subgraph/src/mappings/markets.ts

```diff
@@ -0,0 +1,203 @@
+/* eslint-disable prefer-const */ // to satisfy AS compiler
+
+// For each division by 10, add one to exponent to truncate one significant figure
+import { Address, BigDecimal, BigInt, log } from "@graphprotocol/graph-ts";
+import { Market, Comptroller } from "../../generated/schema";
+import { PriceOracle } from "../../generated/templates/CToken/PriceOracle";
+import { BEP20 } from "../../generated/templates/CToken/BEP20";
+import { CToken } from "../../generated/templates/CToken/CToken";
+
+import {
+  exponentToBigDecimal,
+  mantissaFactor,
+  mantissaFactorBD,
+  zeroBD,
+} from "./helpers";
+
+const CNativeAddr = "0x54dfae480e33dc2befd42caa26a432b11b5a27bd";
+// Used for all cBEP20 contracts
+function getTokenPrice(
+  blockNumber: BigInt,
+  ctokenAddress: Address
+): BigDecimal {
+  let comptroller = Comptroller.load("1");
+  if (!comptroller) {
+    return BigDecimal.zero();
+  }
+  if (!comptroller.priceOracle) {
+    return BigDecimal.zero();
+  }
+  let oracleAddress = Address.fromBytes(comptroller.priceOracle!);
+  let underlyingPrice: BigDecimal;
+
+  /* PriceOracle2 is used from starting of Comptroller.
+   * This must use the cToken address.
+   *
+   * Note this returns the value without factoring in token decimals and wei, so we must divide
+   * the number by (bnbDecimals - tokenDecimals) and again by the mantissa.
+   */
+  // let mantissaDecimalFactor = 18 - underlyingDecimals + 18
+  let oracle = PriceOracle.bind(oracleAddress);
+  let underlyingCall = oracle.try_getUnderlyingPrice(ctokenAddress);
+
+  if (underlyingCall.reverted) {
+    underlyingPrice = zeroBD;
+  } else {
+    underlyingPrice = underlyingCall.value.toBigDecimal().div(mantissaFactorBD);
+  }
+
+  return underlyingPrice;
+}
+
+export function createMarket(
+  marketAddress: string,
+  blockNumber: BigInt
+): Market {
+  let market: Market;
+  let contract = CToken.bind(Address.fromString(marketAddress));
+  // It is cCELO, which has a slightly different interface
+  const decimals = contract.try_decimals();
+  const underlyingPrice = getTokenPrice(
+    blockNumber,
+    Address.fromString(marketAddress)
+  );
+  if (marketAddress == CNativeAddr) {
+    market = new Market(marketAddress);
+    market.underlyingAddress = Address.fromString(marketAddress);
```

Comment:
it should be `market.underlyingAddress = contract.underlying()`, we mark the ETH address as the 0 address in our point system